app/services/users/archive_by_email.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Users
+  class ArchiveByEmail
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    attribute :user
+
+    def call
+      if user_with_matching_email
+        ApplicationRecord.transaction do
+          move_applications(from_user: user_with_matching_email, to_user: user)
+          if Feature.ecf_api_disabled?
+            move_participant_id_changes(from_user: user_with_matching_email, to_user: user)
+            user.participant_id_changes.find_or_create_by!(from_participant_id: user_with_matching_email.ecf_id, to_participant_id: user.ecf_id)
+          end
+        end
+        Rails.logger.info("Archiving user with clashing email address ID=#{user_with_matching_email.id}")
+        Users::Archiver.new(user: user_with_matching_email).archive!
```

Comment:
just incase before this could we add:
```
user.reload
user_with_matching_email.reload
```
we hit issues when we tried to update records before so just incase :)

Reply:
I've added the code to reload `user_with_matching_email`.
For `user` - I've made this attribute private, so it doesn't need to be reloaded.
I can see in other services, the attributes need to be reloaded, because they are accessed outside the service, once the service has been called - I don't think that kind of access is needed in this service.